### DISCUSSION TITLE
Question: Pulsar IO JDBC Java class example (Clickhouse)

### DISCUSSION THREAD

**Discussion by Redirts** [apache/pulsar#18872] [Q&A] [answered]
From what I understand it is possible to create a custom Java class that reads from pulsar and writes via JDBC to a database (in my I would like it to be clickhouse).

https://github.com/apache/pulsar/blob/master/pulsar-io/jdbc/clickhouse/src/main/java/org/apache/pulsar/io/jdbc/ClickHouseJdbcAutoSchemaSink.java

Can anyone provide a complete code example to make this JDBC Sink work?

Thanks

**Comment by jiazhai** [comment]
@Redirts  is this link help? https://pulsar.apache.org/docs/en/io-quickstart/#connect-pulsar-to-postgresql

**Comment by jiazhai** [comment]
would also like to have @vzhikserg in the thread, who is the author for clickhouse connector.

**Comment by Redirts** [comment]
Hi,

The link you provide describes setting up the jdbc sink without writing any
line of code. I would like to be able to manipulate the data before it is
sent to clickhouse via java code.

Thanks

**Comment by tisonkun** [comment, marked as answer]
@Redirts you can fork and patch the upstream ClickHouse connector for your use case in this situation.
